server/src/main/java/org/opensearch/index/query/RangeQueryBuilder.java

```diff
@@ -545,16 +548,13 @@ protected boolean doEquals(RangeQueryBuilder other) {
             && Objects.equals(format, other.format);
     }
 
-    /**
-     * Returns a list of RangeQueryBuilder whose elements, when combined, form the complement of this range query.
-     * May be null.
-     * @return the complement
-     */
-    public List<RangeQueryBuilder> getComplement() {
+    @Override
+    public List<QueryBuilder> getComplement(QueryShardContext context) {
```

Comment:
If you change the return type of `getComplement` to` List<? extends QueryBuilder>`, you can keep the local variable (`complement`) as a `List<RangeQueryBuilder>` and still return it with an acceptable type.